qa/acceptance-tests/src/test/java/io/camunda/it/client/ProcessInstanceSearchTest.java

```diff
@@ -1420,14 +1422,33 @@ void shouldReturnOnlyMatchingTopLevelWhenOrClauseErrorMessageConflicts() {
   }
 
   @Test
-  void shouldReturnEmptyResultForIncorrectIncidentErrorHashCode() {
+  void shouldReturnProcessInstancesWithMatchingTags() {
+    final var result =
+        camundaClient.newProcessInstanceSearchRequest().filter(f -> f.tags("a", "b")).send().join();
+
+    assertThat(result.items()).isNotEmpty();
+    assertThat(result.items())
+        .allSatisfy(
+            item -> {
+              assertThat(item.getTags()).contains("a", "b");
+            });
+  }
+
+  @Test
+  void shouldReturnProcessInstancesWithMatchingTagsAndOtherFilter() {
```

Comment:
🔧 Just to be fully sure, can you add another case where you search just for a single tag ?